app/admin/ServersideAbTest.scala

```diff
@@ -0,0 +1,22 @@
+package admin
+import cookies.ServersideAbTestCookie
+import play.api.mvc.RequestHeader
+
+import scala.util.Random
+
+object ServersideAbTest {
+  // Serverside A/B tests currently only support a single concurrent test
+  // running to 100% audience with a 50%/50% split
+  def getParticipation(implicit request: RequestHeader): Participation = {
+    ServersideAbTestCookie.get.fold(computeParticipation)(_.value match {
```

Comment:
do we need to flatMap with the match statement first and then getOrElse?  otherwise if the cookie is corrupted somehow we will have a match error or something

Reply:
```scala
  def getParticipation(implicit request: RequestHeader): Participation = {
    ServersideAbTestCookie.get.flatMap(_.value match {
      case "Control" => Some(Control)
      case "Variant" => Some(Variant)
      case _ => None
    }).getOrElse(computeParticipation)
  }
```

Reply:
yes you're quite right!